Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift

```diff
@@ -244,3 +254,13 @@ extension _ChatChannelListVC: _ChatChannelListControllerDelegate {
         )
     }
 }
+
+extension _ChatChannelListVC: DataControllerStateDelegate {
+    public func controller(_ controller: DataController, didChangeState state: DataController.State) {
+        if self.controller.channels.isEmpty {
+            loadingIndicator.startAnimating()
+        } else {
+            loadingIndicator.stopAnimating()
+        }
```

Comment:
We need to check the actual state here. Like this will just keep spinning if there are no channels. We have the following cases:
```
/// The controller is created but no data fetched.
case initialized
/// The controllers already fetched local data if any.
case localDataFetched
/// The controller failed to fetch local data.
case localDataFetchFailed(ClientError)
/// The controller fetched remote data.
case remoteDataFetched
/// The controller failed to fetch remote data.
case remoteDataFetchFailed(ClientError)
```
The first two should cause the indicator to be visible + the `isEmpty` check, the rest should hide the indicator.